packages/sdk-react/src/components/receivables/InvoiceDetails/CreateReceivable/sections/ItemsSection.tsx

```diff
@@ -300,50 +306,80 @@ export const ItemsSection = ({
-                          >
-                            <Select
-                              {...field}
-                              id={field.name}
-                              labelId={field.name}
-                              size="small"
-                              defaultValue=""
-                              MenuProps={{ container: root }}
-                              onChange={(e) => {
-                                const vatRateId = e.target.value;
-                                const vatRate = vatRates?.data.find(
-                                  (vatRate) => vatRate.id === vatRateId
-                                );
+                      {isUSEntity ? (
+                        <Controller
+                          name={`line_items.${index}.tax_rate_value`}
+                          control={control}
+                          render={({ field, fieldState: { error } }) => (
+                            <FormControl
+                              variant="outlined"
+                              fullWidth
+                              required
+                              error={Boolean(error)}
+                            >
+                              <TextField
+                                {...field}
+                                id={field.name}
+                                type="number"
+                                inputProps={{ min: 1, max: 100 }}
+                                size="small"
+                                fullWidth={false}
+                                error={Boolean(error)}
+                                helperText={error?.message}
+                                InputProps={{
+                                  endAdornment: '%',
+                                }}
+                                sx={{ minWidth: 100 }}
+                              />
+                            </FormControl>
+                          )}
+                        />
+                      ) : (
+                        <Controller
+                          name={`line_items.${index}.vat_rate_id`}
+                          control={control}
+                          render={({ field, fieldState: { error } }) => (
+                            <FormControl
+                              variant="outlined"
+                              fullWidth
+                              required
+                              error={Boolean(error)}
+                            >
+                              <Select
+                                {...field}
+                                id={field.name}
+                                labelId={field.name}
+                                size="small"
+                                defaultValue=""
+                                MenuProps={{ container: root }}
+                                onChange={(e) => {
+                                  const vatRateId = e.target.value;
```

Comment:
I encourage you not to create a new variable if it’s only used once, as it allocates additional memory unnecessarily. Currently, you’re using it only in `(vatRate) => vatRate.id === vatRateId`. Using it directly as `(vatRate) => vatRate.id === e.target.value` is perfectly fine and is a micro-optimization

Reply:
That was a pre-existing code, but I am happy to modify it as I agree with your comment.

Reply:
Done